### PR TITLE
fomatting and linking the FAQ

### PR DESCRIPTION
**Pull request by mattip**
Follow on to #5374 in order to
- Reformat the page and get rid of the warnings and errors, hopefully without changing the content
- Add a top heading
- Link the FAQ to the userguide index

### CODE REVIEW

**Comment by h-vetinari** [comment]
I avoided introducing linebreaks because that felt quite intrusive compared to the original, even though I agree it's a good idea. In the context of this PR, it kinda overwhelms the other changes, but I think improvement on that file is welcome even so. 👍

File: docs/src/userguide/index.rst [hunk 1]
```diff
@@ -28,6 +28,7 @@ Contents:
    numpy_pythran
    nogil
    troubleshooting
+   FAQ <faq>
```
Comment (scoder) [inline docs/src/userguide/index.rst:31]: I don't think we need to specify the title here. It just works everywhere else.
```suggestion
   faq
```

**Review by scoder — APPROVED** [review]
It's a new file, I don't mind the reformatting.

**Comment by scoder** [comment]
Thanks

**Pull request merged**